src/client/hooks/useTheses.ts

```diff
@@ -5,23 +5,31 @@ import { ThesisData } from '@backend/types'
 
 import apiClient from '../util/apiClient'
 
-interface UseThesesOptions {
+interface UseThesesParams {
   onlySupervised: boolean
+  offset: number
+  limit: number
 }
-export const useTheses = ({ onlySupervised }: UseThesesOptions) => {
-  const queryKey = ['theses', onlySupervised]
-
-  const queryFn = async (): Promise<ThesisData[]> => {
-    const { data } = await apiClient.get(
-      `/theses${onlySupervised ? '?onlySupervised=true' : ''}`
-    )
+export const useTheses = (params: UseThesesParams) => {
+  const queryKey = [
+    'theses',
+    params.onlySupervised,
+    params.offset,
+    params.limit,
+  ]
+
+  const queryFn = async (): Promise<{
+    theses: ThesisData[]
+    totalCount: number
+  }> => {
+    const { data } = await apiClient.get('/theses/paginate', { params })
 
     return data
   }
 
-  const { data: theses, ...rest } = useQuery({ queryKey, queryFn })
+  const { data, ...rest } = useQuery({ queryKey, queryFn })
 
-  return { theses, ...rest }
+  return { ...data, ...rest }
```

Comment:
Just for code readability, should we have here `theses` and `totalCount` extracted explicitly?
Now, needs to go to the backend code to see what values are returned.

Or maybe we could add a generics type to the `useQuery` call above?